src/back-end/lib/routers/auth.ts

```diff
@@ -135,7 +135,56 @@ async function makeRouter(connection: Connection): Promise<Router<any, any, any,
           return makeAuthErrorRedirect(request);
         }
       })
-    }
+    },
+    // /auth/createsession is for testing only, so only exists outside of production
+    // @ts-ignore
+    ...(process.env.NODE_ENV === 'development' ? [{
```

Comment:
Is this enough to ensure this endpoint will never exist in prod?

Reply:
No, this means that your endpoint will only exist in your development build. When testing with cypress you want to test against your production build.
`NODE_ENV` is only telling you whether you're looking at a development build (enabling, hot module reloading, react dev tools, etc) or a production build, it is not related to your dev/test/prod environment.

What you want here is an opt-in feature flag, e.g. `ENABLE_MOCK_AUTH`, and you'll want to run your server with that flag when starting it for your cypress tests ([example](https://github.com/bcgov/cas-cif/blob/07ce72230b0fbccd4ceeae90a279e513dae9975e/app/server/args.ts)).

If your application container is aware of which namespace it's running in, e.g. via a `NAMESPACE` env variable, you could also ensure that this feature flag cannot be enabled when the application is deployed on OpenShift (no matter the namespace).
e.g. 
```ts
export const ENABLE_MOCK_AUTH = process.argv.includes("ENABLE_MOCK_AUTH") && process.env.NAMESPACE === undefined;
```

Reply:
Yes, I'd say this is safe. Tech debt here is we eventually want a validated configuration (eg. [convict](https://github.com/mozilla/node-convict), among many others) but this is good for now.

Reply:
https://github.com/button-inc/digital_marketplace/issues/120